Make allocator check C10_UNLIKELY

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78801

This popped up while having a look at posible causes for https://github.com/pytorch/pytorch/issues/78800

lezcano committed:
Make allocator check C10_UNLIKELY

This popped up while having a look at posible causes for https://github.com/pytorch/pytorch/issues/78800

[ghstack-poisoned]

Lezcano commented on c10/core/impl/alloc_cpu.cpp:
```diff
@@ -58,27 +58,29 @@ void* alloc_cpu(size_t nbytes) {
   void* data;
 #ifdef __ANDROID__
   data = memalign(gAlignment, nbytes);
+  CAFFE_ENFORCE(
+      data,
+      "DefaultCPUAllocator: not enough memory: you tried to allocate ",
+      nbytes,
+      " bytes.");
```
These checks are moved inside the if-elifs blocks as the `else` branch already assures that `datta != nullptr` if `err == 0`.

ezyang commented:
Lint

lezcano committed:
Update on "Make allocator check C10_UNLIKELY"


This popped up while having a look at posible causes for https://github.com/pytorch/pytorch/issues/78800

[ghstack-poisoned]

Lezcano commented:
@pytorchbot merge -g

Lezcano commented:
@pytorchbot merge

suo commented:
You are based on a super old master, and so the CI is not running some jobs which the mergebot requires

pytorchmergebot committed:
Update on "Make allocator check C10_UNLIKELY"


This popped up while having a look at posible causes for https://github.com/pytorch/pytorch/issues/78800

[ghstack-poisoned]

Lezcano commented:
@pytorchbot merge

Lezcano commented:
@pytorchbot merge -g
